java/src/main/java/org/whispersystems/libsignal/state/IdentityKeyStore.java

```diff
@@ -16,67 +14,14 @@
  */
 public interface IdentityKeyStore {
 
-  public enum Direction {
-    SENDING, RECEIVING
-  }
-
-  /**
-   * Get the local client's identity key pair.
-   *
-   * @return The local client's persistent identity key pair.
-   */
-  public IdentityKeyPair getIdentityKeyPair();
-
-  /**
-   * Return the local client's registration ID.
-   * <p>
-   * Clients should maintain a registration ID, a random number
-   * between 1 and 16380 that's generated once at install time.
-   *
-   * @return the local client's registration ID.
-   */
-  public int             getLocalRegistrationId();
-
-  /**
-   * Save a remote client's identity key
-   * <p>
-   * Store a remote client's identity key as trusted.
-   *
-   * @param address     The address of the remote client.
-   * @param identityKey The remote client's identity key.
-   * @return True if the identity key replaces a previous identity, false if not
-   */
-  public boolean         saveIdentity(SignalProtocolAddress address, IdentityKey identityKey);
```

Comment:
Signal has a need to associate trusted identity keys with addresses because a given address can have different identity keys, especially over time. Since our address contains the identity key, we have no need for this mechanism.